### PR TITLE
Move GeneratorStartup extension to powsybl-core

### PR DESCRIPTION
**Pull request by miovd**
Signed-off-by: VEDELAGO MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Moving code

**What is the current behavior?** *(You can also link to an open issue here)*
`GeneratorStartup` is in [powsybl/powsybl-network-hypothesis](https://github.com/powsybl/powsybl-network-hypothesis)

**What is the new behavior (if this is a feature change)?**
`GeneratorStartup` is in core

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
@miovd @annetill attribute naming seems not to bad?

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/GeneratorStartup.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.powsybl.iidm.network.Generator;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public interface GeneratorStartup extends Extension<Generator> {
+
+    @Override
+    default String getName() {
+        return "generatorStartup";
```
Comment (miovd) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/GeneratorStartup.java:19]: I would change it back to `startup` to be consistent with other extension name (see `LoadDetail` for example). The extension name should not contain the equipment.

Comment (geofjamg) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/GeneratorStartup.java]: Be careful to keep compat with existing data

**Comment by miovd** [comment]
> > > I changed the PR to WIP, to let time to answer an important question: we are not able to load again xiidm from CVG because everything has been renamed. Is it really ok? Maybe we should keep compat in xml read?
> > 
> > 
> > You're right! There has been two renamed attributes, I will add the backward compatibility
> 
> This is also incompatible because name of the extension has changed. We should add a unit test with old xml to ensure it is correctly imported.

The extension's name has not changed, I reverted Anne's changes, it is still `startup`

**Comment by miovd** [comment]
> > > > > I changed the PR to WIP, to let time to answer an important question: we are not able to load again xiidm from CVG because everything has been renamed. Is it really ok? Maybe we should keep compat in xml read?
> > > > 
> > > > 
> > > > You're right! There has been two renamed attributes, I will add the backward compatibility
> > > 
> > > 
> > > This is also incompatible because name of the extension has changed. We should add a unit test with old xml to ensure it is correctly imported.
> > 
> > 
> > The extension's name has not changed, I reverted Anne's changes, it is still `startup`
> 
> Ah ok... so it was an old mistake, generatorStartup would have been more meaningful. To fix this, we could support some extension name aliases in `ExtensionXmlSerializer` to allow extension renaming.

As I said above, is it what we want? All other extensions don't have the extendable type in their name. Or do we want to change everywhere?

**Review by geofjamg — APPROVED** [review]

**Pull request merged**